x-pack/test/functional/apps/search_playground/playground_overview.ess.ts

```diff
@@ -5,90 +5,322 @@
+  // const commonAPI = MachineLearningCommonAPIProvider(ftrContext);
   const supertest = getService('supertest');
   const esArchiver = getService('esArchiver');
+
+  const log = getService('log');
+
   const createIndex = async () => await esArchiver.load(esArchiveIndex);
-  let removeOpenAIConnector: () => Promise<void>;
-  const createConnector = async () => {
-    removeOpenAIConnector = await createOpenAIConnector({
-      supertest,
-      requestHeader: commonAPI.getCommonRequestHeader(),
-    });
-  };
 
-  describe('Playground Overview', () => {
+  async function deleteConnectors() {
+    const response = await supertest.get('/api/actions/connectors').expect(200);
+    for (const connector of response.body) {
+      if (connector.connector_type_id === '.gen-ai') {
+        await supertest
+          .delete(`/api/actions/connector/${connector.id}`)
+          .set('kbn-xsrf', 'foo')
+          .expect(204);
+      }
+    }
+  }
+
+  describe('Playground', () => {
+    let proxy: LlmProxy;
     before(async () => {
+      await deleteConnectors();
+      proxy = await createLlmProxy(log);
       await pageObjects.common.navigateToApp('enterpriseSearchApplications/playground');
     });
 
     after(async () => {
       await esArchiver.unload(esArchiveIndex);
-      await removeOpenAIConnector?.();
+      await deleteConnectors();
+      proxy.close();
+      // await removeOpenAIConnector?.();
     });
 
-    describe('start chat page', () => {
-      it('playground app is loaded', async () => {
-        await pageObjects.searchPlayground.PlaygroundStartChatPage.expectPlaygroundStartChatPageComponentsToExist();
+    describe('setup Page', () => {
+      it('is loaded successfully', async () => {
         await pageObjects.searchPlayground.PlaygroundStartChatPage.expectPlaygroundHeaderComponentsToExist();
+        await pageObjects.searchPlayground.PlaygroundStartChatPage.expectPlaygroundStartChatPageComponentsToExist();
       });
 
-      it('show no index callout', async () => {
-        await pageObjects.searchPlayground.PlaygroundStartChatPage.expectNoIndexCalloutExists();
-        await pageObjects.searchPlayground.PlaygroundStartChatPage.expectCreateIndexButtonToExists();
-      });
+      describe('without gen ai connectors', () => {
+        it('should display the set up connectors button', async () => {
+          await pageObjects.searchPlayground.PlaygroundStartChatPage.expectAddConnectorButtonExists();
+        });
 
-      it('hide no index callout when index added', async () => {
-        await createIndex();
-        await pageObjects.searchPlayground.PlaygroundStartChatPage.expectSelectIndex(indexName);
+        it('creates a connector successfully', async () => {
+          await pageObjects.searchPlayground.PlaygroundStartChatPage.expectOpenConnectorPagePlayground();
+          await pageObjects.searchPlayground.PlaygroundStartChatPage.createOpenAIConnector(proxy);
```

Comment:
Need to create an api function for creating connector as we don't need to test third-party library. Because if they changed name of field it will broke our test.

Reply:
Yeah I agree that we need to test how flyout is opening, because this is related to our code, but testing the flyout seems like an antipattern “Third-Party Code Testing.” Which can lead to:
- Unnecessary Maintenance: Your tests might fail if the third-party library updates or changes, leading to maintenance work unrelated to your actual code.
- Redundancy: Most third-party libraries come with their own tests. Testing them again in your integration suite duplicates effort without additional value.
- False Confidence: Passing tests might give a false sense of security about the library’s behavior under all conditions, which isn’t practical to ensure comprehensively.